packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -275,23 +292,16 @@ function getValidTelemetryProps(obj: any, keysToOmit: Set<string>): ITelemetryPr
             continue;
         }
         const val = obj[key];
-        switch (typeof val) {
-            case "string":
-            case "number":
-            case "boolean":
-            case "undefined":
-                props[key] = val;
-                break;
-            default: {
-                if (isTaggedTelemetryPropertyValue(val)) {
-                    props[key] = val;
-                } else {
-                    // We don't support logging arbitrary objects
-                    props[key] = "REDACTED (arbitrary object)";
-                }
-                break;
-            }
+        if (isTaggedTelemetryPropertyValue(val)) {
+            props[key] = val;
+        } else if (parseJSONObject(val) !== null) {
+            props[key] = parseJSONObject(val) as TelemetryEventPropertyType;
```

Comment:
Can we avoid calling this twice?  Then you won't need the `as` either after the null check.